include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -683,6 +783,123 @@ __pattern_transform_scan_multi_group(_ExecutionPolicy&& __exec, _InRng&& __in_rn
         unseq_backend::__global_scan_functor<_Inclusive, _BinaryOperation, _InitType>{__binary_op, __init});
 }
 
+template <typename _SizeType>
+struct __invoke_single_group_copy_if
+{
+    // Specialization for devices that have a max work-group size of at least 1024
+    static constexpr ::std::uint16_t __targeted_wg_size = 1024;
```

Comment:
Did you experiment with the work group size? A smaller size could suffice to saturate the memory bandwidth but might reduce the complexity of the group scan.

Reply:
This size was chosen for scan, but I did not do experiments specifically for `copy_if`. I will experiment a bit with these parameters and get back to you.